staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -672,451 +674,394 @@ func TestApfWithRequestDigest(t *testing.T) {
-		} else if !(secondReqResult.response.StatusCode == http.StatusTooManyRequests || secondReqResult.response.StatusCode == http.StatusGatewayTimeout) {
-			t.Errorf("Expected HTTP status code: %d or %d for request: %q, but got: %#+v", http.StatusTooManyRequests, http.StatusGatewayTimeout, secondRequestEnqueuedPath, secondReqResult.response)
-		}
+type enqueuedRequestTimingOut struct {
+	firstReqVerifier, secondReqVerifier verifier
+}
 
-		close(stopCh)
-		t.Log("Waiting for the controller to shutdown")
+func (test enqueuedRequestTimingOut) runT(t *testing.T) {
+	// scenario:
+	// a) priority level concurrency is set to 1, and queue length is 1
+	// b) the first request arrives and is being executed
+	// c) the second request arrives and is is enqueued
+	// d) the first request handler blocks indefinitely
+	// e) the second request should be rejected by APF
+	// f) the first request eventually times out
+	const (
+		userName                                                           = "alice"
+		serverConcurrency, plConcurrencyShares, plConcurrency, queueLength = 1, 1, 1, 1
+	)
 
-		controllerErr := <-controllerCompletedCh
-		if controllerErr != nil {
-			t.Errorf("Expected no error from the controller, but got: %s", fmtError(controllerErr))
+	fsName, plName := newFSandPLNames()
+	apfConfiguration := newConfiguration(fsName, plName, userName, plConcurrencyShares, queueLength)
+	stopCh := make(chan struct{})
+	controller, controllerCompletedCh := startAPFController(t, stopCh, apfConfiguration, serverConcurrency, plName, plConcurrency)
+
+	firstRequestTimesOutPath, secondRequestEnqueuedPath := "/request/first/time-out-as-designed", "/request/second/enqueued-as-designed"
+	firstReqHandlerErr := errSender{who: firstRequestTimesOutPath, ch: make(chan error, 1)}
+	firstReqInProgressCh := make(chan struct{})
+	firstReqRoundTripDoneCh, secondReqRoundTripDoneCh := make(chan struct{}), make(chan struct{})
+	requestHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		case r.URL.Path == firstRequestTimesOutPath:
+			defer close(firstReqHandlerErr.ch)
+
+			close(firstReqInProgressCh)
+			<-firstReqRoundTripDoneCh
+
+			// make sure we wait until the caller of the second request returns, this is to
+			// ensure that second request never has a chance to be executed (to avoid flakes)
+			<-secondReqRoundTripDoneCh
+
+			// we expect the timeout handler to have timed out this request by now and any attempt
+			// to write to the response should return a http.ErrHandlerTimeout error.
+			_, firstRequestInnerHandlerWriteErr := w.Write([]byte("foo"))
+			firstReqHandlerErr.ch <- firstRequestInnerHandlerWriteErr
+
+		case r.URL.Path == secondRequestEnqueuedPath:
+			// we expect the concurrency to be set to 1 and so this request should never be executed.
+			t.Errorf("Expected second request to be enqueued: %q", secondRequestEnqueuedPath)
```

Comment:
I would say "Expected second request to be rejected from queue rather than executed"

Reply:
yes, fixed